extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthorizer.java

```diff
@@ -18,17 +24,28 @@ public class HttpAuthorizer extends AbstractHttpAuthorizer {
 
     HttpAuthorizer(HttpAuthenticator httpAuthenticator, IdentityProviderManager identityProviderManager,
             AuthorizationController controller, Instance<HttpSecurityPolicy> installedPolicies,
-            BlockingSecurityExecutor blockingExecutor) {
-        super(httpAuthenticator, identityProviderManager, controller, toList(installedPolicies), blockingExecutor);
+            BlockingSecurityExecutor blockingExecutor, BeanManager beanManager,
+            Event<AuthorizationFailureEvent> authZFailureEvent, Event<AuthorizationSuccessEvent> authZSuccessEvent,
+            @ConfigProperty(name = "quarkus.security.events.enabled") boolean securityEventsEnabled) {
+        super(httpAuthenticator, identityProviderManager, controller, toList(installedPolicies), beanManager, blockingExecutor,
+                authZFailureEvent, authZSuccessEvent, securityEventsEnabled);
     }
 
     private static List<HttpSecurityPolicy> toList(Instance<HttpSecurityPolicy> installedPolicies) {
         List<HttpSecurityPolicy> globalPolicies = new ArrayList<>();
         for (HttpSecurityPolicy i : installedPolicies) {
-            if (i.name() == null) {
+            if (i.name() == null && isNotPathMatchingPolicyWithoutPerms(i)) {
```

Comment:
It is a very minor update but since you are planning to work on HttpSecurityPolicy related PR later as well, then I propose to deal with this update there. I'm not sure why a policy without name can still be applied, but IMHO, in any case, it is another conversation

Reply:
> It is a very minor update but since you are planning to work on HttpSecurityPolicy related PR later as well, then I propose to deal with this update there.

It is just something I mentioned thanks to security events. I think this is just logical change as you don't want to check paths only to find out there are none.

It is not related to what I'll do in future, but I can create separate PR if you like and drop this change.

>  I'm not sure why a policy without name can still be applied, but IMHO, in any case, it is another conversation

this is about `io.quarkus.vertx.http.runtime.security.AbstractPathMatchingHttpSecurityPolicy` being run when there are no HTTP permissions. Why does that bean exist? Because whether bean is going to be created is decided during build time, but HTTP permissions are runtime defined.

Reply:
@michalvavrik Sure, looks like it is worth backporting, so if it stays in this PR then it won't be backported, so it is up to you

Reply:
I'll move it.